serverless/lambda/src/main/java/uk/gov/di/lambdas/UpdateProfileHandler.java

```diff
@@ -81,72 +84,99 @@ public APIGatewayProxyResponseEvent handleRequest(
-                    authenticationService.updatePhoneNumber(
-                            profileRequest.getEmail(), profileRequest.getProfileInformation());
-                    sessionService.save(session.get().setState(ADDED_UNVERIFIED_PHONE_NUMBER));
-                    LOGGER.info(
-                            "Phone number updated and session state changed. Session state {}",
-                            ADDED_UNVERIFIED_PHONE_NUMBER);
-                    return generateApiGatewayProxyResponse(
-                            200, new BaseAPIResponse(session.get().getState()));
+                    {
+                        validateStateTransition(session.get(), ADDED_UNVERIFIED_PHONE_NUMBER);
+                        authenticationService.updatePhoneNumber(
+                                profileRequest.getEmail(), profileRequest.getProfileInformation());
+                        sessionService.save(session.get().setState(ADDED_UNVERIFIED_PHONE_NUMBER));
+                        LOGGER.info(
+                                "Phone number updated and session state changed. Session state {}",
+                                ADDED_UNVERIFIED_PHONE_NUMBER);
+                        return generateApiGatewayProxyResponse(
+                                200, new BaseAPIResponse(session.get().getState()));
+                    }
                 case CAPTURE_CONSENT:
-                    Optional<ClientSession> clientSession =
-                            clientSessionService.getClientSessionFromRequestHeaders(
-                                    input.getHeaders());
+                    {
+                        Optional<ClientSession> clientSession =
+                                clientSessionService.getClientSessionFromRequestHeaders(
+                                        input.getHeaders());
+
+                        if (!clientSession.isPresent()) {
+                            LOGGER.info("ClientSession not found.");
+                            return generateApiGatewayProxyErrorResponse(
+                                    400, ErrorResponse.ERROR_1000); // 1020
+                        }
+
+                        try {
+                            AuthenticationRequest authenticationRequest =
+                                    AuthenticationRequest.parse(
+                                            clientSession.get().getAuthRequestParams());
+                            clientId = authenticationRequest.getClientID().getValue();
+                        } catch (ParseException e) {
+                            LOGGER.info(
+                                    "Cannot retreive auth request params from client session id.");
+                            return generateApiGatewayProxyErrorResponse(
+                                    400, ErrorResponse.ERROR_1001);
+                        }
+
+                        Optional<Map<String, List<String>>> clientConsents =
```

Comment:
this doesn't look like the right object is being persisted

Reply:
Yeah your right. We discussed this on Friday, this section was part of the capture consent ticket me and Dan were working on previously, the model does need updating but it is not part of the scope of this ticket and I will do the refactoring when I revisit that. It is green in the diff probably due to IntelliJ formatting code rather than it being a new addition.